move to rust 2021

NukeManDan opened the pull request:
Confirmed to build locally on Ubuntu 20.04.3 LTS:
```
rustc 1.56.1 (59eed8a2a 2021-11-01)
rustc 1.58.0-nightly (e90c5fbbc 2021-11-12)
```

ordian reviewed (approved):
Thanks! It would be good to document in the changelogs that the minimum required rust version (MSVR) is now bumped to 1.56.

NukeManDan commented:
> Thanks! It would be good to document in the changelogs that the minimum required rust version (MSVR) is now bumped to 1.56.

Updated and added MSRV 1.56.1, for discussion on why: https://github.com/libp2p/rust-libp2p/pull/2339
Also removed the `reslover = 2` as that's no longer needed, and for a few missing CHANGELOG files, now added to all crates and all crates are 2021 + 1.56.1

ordian commented on uint/fuzz/CHANGELOG.md:
```diff
@@ -0,0 +1,8 @@
+[Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
+
+## [Unreleased]
+- Migrated to 2021 edition, enforcing MSRV of `1.56.1`. [#601](https://github.com/paritytech/parity-common/pull/601)
```
this crate is not going to be published
